.github/workflows/test_pkgci_regression_run.yml

```diff
@@ -0,0 +1,203 @@
+# Copyright 2023 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+name: PkgCI Regression Test
+on:
+  workflow_call:
+    inputs:
+      artifact_run_id:
+        type: string
+        default: ""
+  workflow_dispatch:
+    inputs:
+      artifact_run_id:
+        type: string
+        default: ""
+
+jobs:
+  test_regression_suite:
+    name: "test_regression_suite :: ${{ matrix.name }}"
+    runs-on: ${{ matrix.runs-on }}
+    strategy:
+      fail-fast: false
+
+      # Note: these jobs should use persistent runners with local caches.
+      # Downloading test files (50GB+) without a cache can take 20+ minutes.
+      matrix:
+        include:
+          # CPU
+          - name: cpu_llvm_task
+            backend: cpu
+            runs-on:
+              - self-hosted # must come first
+              - persistent-cache
+              - Linux
+              - X64
+
+          # AMD GPU
+          - name: amdgpu_rocm_mi250_gfx90a
+            rocm-chip: gfx90a
+            backend: rocm
+            sku: mi250
+            runs-on: nodai-amdgpu-mi250-x86-64
+          - name: amdgpu_rocm_mi300_gfx942
+            rocm-chip: gfx942
+            backend: rocm
+            sku: mi300
+            runs-on: nodai-amdgpu-mi300-x86-64
+          - name: amdgpu_rocm_mi308_gfx942
+            rocm-chip: gfx942
+            backend: rocm
+            sku: mi308
+            runs-on: nodai-amdgpu-mi308-x86-64
+    env:
+      PACKAGE_DOWNLOAD_DIR: ${{ github.workspace }}/.packages
+      IREE_TEST_PATH_EXTENSION: ${{ github.workspace }}/build_tools/pkgci/external_test_suite
+      VENV_DIR: ${{ github.workspace }}/venv
+      TEST_OUTPUT_ARTIFACTS: ${{ github.workspace }}/model_output_artifacts
+    steps:
+      - name: Checking out IREE Test Suite
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+        with:
+          submodules: false
+      - uses: actions/setup-python@0b93645e9fea7318ecaed2b359559ac225c90a2b # v5.3.0
+        with:
+          # Must match the subset of versions built in pkgci_build_packages.
+          python-version: "3.11"
+      - uses: actions/download-artifact@fa0a91b85d4f404e444e00e005971372dc801d16 # v4.1.8
+        with:
+          name: linux_x86_64_release_packages
+          path: ${{ env.PACKAGE_DOWNLOAD_DIR }}
+      - name: Setup venv
+        run: |
+          ./pkgci_test_suite/build_tools/setup_venv.py ${VENV_DIR} \
+            --artifact-path=${PACKAGE_DOWNLOAD_DIR} \
+            --fetch-gh-workflow=${{ inputs.artifact_run_id }}
+          source ${VENV_DIR}/bin/activate
+          pip install -e pkgci_test_suite
```

Comment:
A fork of `setup_venv.py` shouldn't be needed in the test suite repository. That's an artifact of how testing is run on pull requests in the IREE repository (one workflow builds the packages from source, then other workflows install those packages). In a separate repository like this one, we would typically just install the already published nightly packages:

https://github.com/iree-org/iree-test-suites/blob/f8f2771c932322a2a813a51a9e6ef71ca7eb1f80/.github/workflows/test_sharktank_models.yml#L45-L55 https://github.com/iree-org/iree-test-suites/blob/f8f2771c932322a2a813a51a9e6ef71ca7eb1f80/sharktank_models/requirements-iree.txt#L1-L9